### ISSUE TITLE
Meetup - ecology in Julia

### ISSUE THREAD

**Issue by mkborregaard** [EcoJulia/SpatialEcology.jl#49]
@tpoisot @richardreeve @kescobo 
I'm currently planning the "VizCon" in Berlin, the second installment of a meeting to advance plotting (and networking) in Julia. Last time it was really successful.

I was thinking, maybe we could all really benefit from doing a 1-week meetup/hackathon to talk and code and get ecology in Julia to a place where it's well-integrated and generally useful? I wouldn't mind coming to Montreal for a week at some point after the summer, which should also be really convenient for Kevin (and Richard loves travelling :smile: ).

What do you say?

**Comment by tpoisot** [comment]
It would be great - at least locally, I know that @gabdansereau & @FrancisBanville are using Julia a lot for spatial ecology things, so we can also involve some grad students. We can also include a one hour seminar for one of you guys.

**Comment by mkborregaard** [comment]
Yeah great, would those students be willing to contribute to building up an ecosystem? That would be pretty awesome

**Comment by tpoisot** [comment]
I think so - @gabrieldansereau is using `SimpleSDMLayers` and moving into questions of endemism, using eBird data, so this is close to the goals of `SpatialEcology`. @francisbanville is working on network prediction in space, which also fits. I think that if we come up with a specific idea leading up to a deliverable (release + draft of the software note), this could be a very productive week.

**Comment by FrancisBanville** [comment]
I would be very happy to contribute to the ecosystem! 

I started my PhD 5 months ago with @tpoisot and I think this project is a great opportunity for me to share and improve my developing expertise in Julia and spatial network ecology.

**Comment by gabrieldansereau** [comment]
I would be happy to contribute too!

As Timothee said, I am already using some of the packages for spatial ecology. A well-integrated ecosytem could be very useful, and I'd love to help develop it.

**Comment by mkborregaard** [comment]
Awesome! I'm in. Now we just need @kescobo and @richardreeve too and find a date :-)

**Comment by mkborregaard** [comment]
September and October are best for me - end August might work in a cinch

**Comment by kescobo** [comment]
Oh yeah - Montreal would be awesome. Need to check in with family about dates, but I think I could make it happen. I think Sept/Oct would be easier for me as well.

**Comment by mkborregaard** [comment]
Maybe also interesting for @rafaqz

**Comment by rafaqz** [comment]
Thanks that sounds great, this has been something I've been meaning to get involved in for a while, seeing I'm working on a lot ecology/spatial packages.

I should be in Portugal for Juliacon, but not sure I could make another trip to Europe this year (I'm in Melbourne)

**Comment by kescobo** [comment]
I suppose from the perspective of an Aussie, Montreal could be considered Europe :-P

**Comment by rafaqz** [comment]
Oh I just read the first line about Berlin above, Motreal is worse lol

**Comment by richardreeve** [comment]
Sorry about missing this earlier - I'm certainly interested, and @claireh93 might be too. I'm not a mad keen traveller, so Berlin sounds a lot more tempting than Montreal, but I'm persuadable! October starts to get difficult for me, but September should be fine...

**Comment by richardreeve** [comment]
Could people potentially move on to somewhere else after JuliaCon if they were coming to that?

**Comment by claireh93** [comment]
Yes this sounds good, I'm interested too! I would probably be able to make it for some/most of this, especially if it's somewhere in Europe.

**Comment by mkborregaard** [comment]
Great! I'm not coming to JuliaCon. I've got a little bit of a hidden agenda on Montreal, because I'd like to spend a week collaborating with another good colleague who's at McGill and thought I could back-to-back them. I'm doing my best to not fly more than absolutely necessary

**Comment by richardreeve** [comment]
Ah, okay. We'll have to look into sources of funding for drifting across the Atlantic. The current work we're doing is climate change related, and we had to promise not to spend any of the money on bad things like flying! We're even carbon offsetting our simulations... once we work out how to :)

**Comment by richardreeve** [comment]
PS I should say, the reason I mentioned JuliaCon was not because we'd have the meeting there, but because a lot of people might be on the same continent, so if people were free the week before or after, we might be able to meet in Europe somewhere... obviously that doesn't help with the McGill plan though.

**Comment by tpoisot** [comment]
@mkborregaard there's also @eric-pedersen at Concordia, who from what I understand has some interest in using Julia for ecology as well. So we may have a critical mass of people locally to have a very produtive week.

**Comment by kescobo** [comment]
Well, I can't do Europe, which is one of the reasons Montreal appeals to me (if I can drive, my wife and son can come too and that makes it feasible).

**Comment by mkborregaard** [comment]
JuliaCon is smack in the middle of my family holiday. To a Dane it's an absolutely terrible idea to put a work-related conference in the middle of July...

**Comment by mkborregaard** [comment]
@tpoisot great!

**Comment by richardreeve** [comment]
Okay, we'll look into other funding sources... airbnb is suddenly looking v attractive!

**Comment by eric-pedersen** [comment]
I would definitely be interested in a meetup in Montreal; I'm only a Julia novice currently (my expertise is in R), but I'm investing time in getting up to speed, since tools like DifferentialEquations.jl and now SpatialEcology.jl are really at the cutting edge for modelling tools. One of my long-term goals for the lab is to develop a simulation suite for metapopulation and metacommunity dynamics, and Julia seems like the ideal platform for that. 

I'd only be able to make Montreal though. . . I'm also trying to reduce long-distance travel, and I teach in the fall, which makes travel for that long very difficult.

**Comment by richardreeve** [comment]
Okay, I can find money to come over, but the sooner we set dates (and the better they are aligned with cheap days to fly!) the better as far as I and @claireh93 the better...

**Comment by rafaqz** [comment]
Montreal is probably out for me. I might see some of you at JuliaCon - I should be talking about DynamicGrids.jl and Dispersal.jl there, which are also looking like cutting edge ecological modelling tools now, and may be of interest to some of you.

**Comment by mkborregaard** [comment]
What days would be good for you guys then?

**Comment by kescobo** [comment]
Generally for me/my family would be Sunday-Wednesday, and absolutely ideal would be 10/11-10/14. Only exception to that in Sept/Oct is 9/28.

**Comment by kescobo** [comment]
I also have a colleague here at Wellesley (Jakie Matthes) that's julia-curious, and I might be able to get her to come along. In any case, it would be awesome to come back with some concrete suggestions for her to get started.

**Comment by mkborregaard** [comment]
Great - I can do almost any time during those two months at the moment, except for... 10th-18th October, sorry.

**Comment by tpoisot** [comment]
I can't do week-ends, but mostly I will be able to find the time during any week of the fall term (plus or minus one half day of teaching)

**Comment by richardreeve** [comment]
I can’t realistically make after late September unfortunately - I manage our graduate students and most arrive end of September / beginning of October - and there wouldn’t be enough free days to cross the Atlantic and back. Early sept would be best for me...

**Comment by mkborregaard** [comment]
sounds like the first or second week of sept could work for everyone?

**Comment by eric-pedersen** [comment]
The first week of September would work best for me, as classes start here
in the second week. I could probably do the 2nd week as well, though.

**Comment by kescobo** [comment]
First week of September is best for me too - should we lock it in? Sept 6-8?

**Comment by tpoisot** [comment]
yes let's

**Comment by mkborregaard** [comment]
Sunday to Tuesday - why those days? If I go to Montreal I'd like to spend two weeks - I only fly once a year. I could spend a week with another colleague at Concordia (let me double-check he has time), but it would be nice to have a little more time for this project too. How much time could people spare?

**Comment by kescobo** [comment]
>Sunday to Tuesday - why those days? 

[That's just what works for my schedule](https://github.com/EcoJulia/SpatialEcology.jl/issues/49#issuecomment-585252128) (my wife sees clients and any time she takes off is lost income). We'd probably come up Saturday evening and leave Wednesday morning, but I'm totally happy if all y'all go longer in either direction.

**Comment by mkborregaard** [comment]
Ah, OK, great, I thought it might have been a mistake because of the match with the weekend, I'd forgotten about that comment. Sounds good if we can fill in a couple of days around it too as it best fits the others (either direction is fine with me too)

**Comment by tpoisot** [comment]
We can have a longer overall time where not everyone attends all the days as well

**Comment by richardreeve** [comment]
Yes, I definitely wouldn't come for just 3 days, but I could work around 6-8. It'll depend on what's cheapest I imagine... probably mid-week to mid-week?

**Comment by mkborregaard** [comment]
Prices don't vary across the week from CPH - I can fly direct for like 300£ return

**Comment by richardreeve** [comment]
Hmm. I don't live in a transport hub, so there are no direct flights and the prices vary a lot!

**Comment by mkborregaard** [comment]
Well then you suggest the dates :-)

**Comment by richardreeve** [comment]
At the moment Wed-Wed looks good? I'm on strike for most of the next several weeks in a massive dispute on pay and working conditions, so it'll be a while before I get this all booked though and everything could change (hey, I could even be better paid!). Anyway, I'm not too fussed.

**Comment by kescobo** [comment]
So Sept 2nd- Sept 9th?

**Comment by mkborregaard** [comment]
For me and my colleague in Montreal Sun-Sat would work a lot better, i.e. 6-12. That would also put Kevin at the beginning to participate in design decisions. Richard, is there a significant expense difference for you with that?

**Comment by mkborregaard** [comment]
@richardreeve if you're not striking - what's your reaction to that suggestion? Or could be Fri-Fri (4-11) as a compromise?

**Comment by richardreeve** [comment]
Anything is fine tbh. I’ll work around it. I’m on strike for the next 2 weeks and then away the following week, so if you want to sort something out, just go for it!

**Comment by kescobo** [comment]
It looks like my wife might actually be able to take that Saturday (the 4th) off, since it's a holiday weekend in the US, in which case we might be able to come up Friday. But I'm also happy to have a day in Montreal to sight-see... I think if Sunday-Sat works best for @mkborregaard Let's just do that. 

Julia ecology meet up Sept 6th-Sept 12th... going once...

**Comment by mkborregaard** [comment]
Sat to Sat then :-)

**Comment by kescobo** [comment]
Well, it's not for sure... God I'm a pain in the ass 😆

**Comment by richardreeve** [comment]
Not wishing to sound negative, but having come back from northern Italy last week to 2 weeks in isolation (one week to go, even though I have tested negative for coronavirus, because I have a cough!), I'm beginning to wonder exactly how wise it is to plan for a visit anywhere at the moment... I'm supposed to be on at least 5 more international trips this year including Montreal, but I'm struggling to believe they are all (any of them?) going to happen, and I'm certainly not booking any of them just yet...

**Comment by mkborregaard** [comment]
Shit. What were you doing in North Italy? Glad to hear you tested negative.

**Comment by richardreeve** [comment]
I was on long-booked holiday skiing! It was great, but the whole "save the planet by taking a train through England, France and Italy instead of flying" turned out to be a bit misjudged! To be honest, at this point I'd have been happy to get infected, so I could deal with it while we still had a functioning health care system...

**Comment by tpoisot** [comment]
In case the in-person meeting doesn't happen, we can (i) gather the local folks in a room large enough to maintain a 2m distance (probably), and then use a combo of zoom and teletype to work remotely.

**Comment by kescobo** [comment]
I suspect things will be much calmer by end of summer. Possibly gearing up for the next flu/corona season. 

Or else civilization will have collapsed and nothing will matter.

**Comment by mkborregaard** [comment]
I guess @richardreeve really is the virologist here though. Isn't your key research area influenza virus?

**Comment by richardreeve** [comment]
Yes, we just published a PNAS paper on epidemiological and ecological interferences between flu (A and B), the common cold (rhinovirus) and other viruses, but sadly not coronaviruses, otherwise I guess my career would now be made! And we've just finished the vaccine selection for N Hemisphere winter flu vaccines, which will now take 8 months to manufacture... it'll be a while - this time next year at the earliest *I guess* - before we have any hope of a coronavirus vaccine. The good news is that the public health measures being brought in in Asia have stamped on the flu season - in some countries worst affected it's already ended 2 months early!

As far as Sept is concerned though, I just don't think we have enough data. I certainly don't know anything about its seasonality - if it survives better in the summer, that won't be ideal(!). If worse, then you're right, Sept may be too soon for a second wave. If it's pretty insensitive to climate, then it'll all depend on how effective our public health measures are - if they're utterly useless, then the worst may be over by Sept, but whether there'll be enough public infrastructure left in N America to host a meeting is probably in question. If they're averagely good, we could drag out the epidemic wave for months (which is good, we can keep hospitals functioning). If they're fantastic (à la Singapore), and I've seen no evidence they will be (though they did a good job with me to be fair), then maybe we'll have squashed this outbreak by then, so we'll be waiting for a second wave again...

Anyway, **I don't have any more information than anyone else**, so certainly don't believe anything I say here, but I'd just suggest not making any firm plans yet. For instance, my university could very plausibly impose a blanket ban on travel almost irrespective of the severity of the situation to protect our ability to function in the new academic year starting mid-Sept. I believe one of the virology labs I collaborate with have already done that to protect themselves. And if I was to target any specific country except China and Iran with a specific travel ban, it would be the US because of its disastrous public health system (the numbers I saw last night were 17 confirmed dead, only 8 recovered, for a disease with ~3% case fatality rate?! Hopefully that's wrong)... and Montreal is pretty close. Having said that the US obviously has shedloads of money to throw at this if it decides to, so it can easily come back from this if it decides to fund the poor and the sick...

Who knows, fundamentally? On the plus side, flights may be cheap!

**Comment by richardreeve** [comment]
That was a bit longer than I'd planned, sorry... tl;dr I think you're probably right that Sept may be a window of opportunity, but who knows!

**Comment by tpoisot** [comment]
Hi all - let's revive this discussion, do we want to have online sprint instead? It's clear there will be no travel in September, but we can most likely get things done remotely.

**Comment by richardreeve** [comment]
We're in a bit of an odd situation - we're currently using our spatial model as a COVID-19 model for the UK. Sadly, there are too few Julia programmers out there to spend time on our day jobs, much less anything else. Invenia are being very helpful, but if it's still ongoing in September (and it may well be obviously) then this won't be an option. If things have all settled down, then it would be great, but otherwise it'll be impossible to be honest.

**Comment by kescobo** [comment]
I've been meaning to revisit but was still holding out some totally unjustified hope. I think you're right. I'm in for online sprint.

@richardreeve Totally understandable - perhaps we can do a bit more up-front planning so that you can chime in on design decisions and then the rest of us can push ahead with a bit more confidence that we can get everything to work together in the end. 

Also tagging @pdimens, author of https://github.com/pdimens/PopGen.jl, since I think he would make a valuable addition (and integrating population genetics with all of the ecology tooling would be just amazing, I think).

**Comment by pdimens** [comment]
All all for incorporation of PopGen.jl into EcoJulia!

**Comment by tpoisot** [comment]
@richardreeve fully understandable and I agree with @kescobo idea of agreeing on implementation -- having @pdimens onboard is also fantastic! Do we feel like a quick zoom is justified at this point? I know that at least @FrancisBanville and @gabrieldansereau will attend JuliaCon virtually to talk about their ecological use cases, so they might have some additional comments (and pain points they identified).

**Comment by mkborregaard** [comment]
Quick zoom sounds good. Is everyone online at this point?

**Comment by kescobo** [comment]
I'm UTC-4 (US east coast) and available most days ~9am-4pm

**Comment by pdimens** [comment]
Same as @kescobo -1

**Comment by FrancisBanville** [comment]
Yes, I agree that a quick Zoom call can be helpful! 

I will be available as of August 10th, almost at any time during the day.

**Comment by tpoisot** [comment]
Same as Francis

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by kescobo** [comment]
Alright, just to move things along, I'm going to propose next Tuesday Aug 11, UTC 15:00. 

Going to post zoom details on slack in #ecology channel, shoot me an e-mail (there's an address on my GH profile) or DM on Discourse if you're not using slack.

**Comment by tpoisot** [comment]
Do you mean #ecojulia? I don't see an #ecology channel on the main slack. Should we setup a gitter?

**Comment by kescobo** [comment]
I've set up an [ecology zulip stream](https://julialang.zulipchat.com/#streams/252417/Ecology) for more persistent conversation. 

From slack discussion - we'll meet today as I proposed, but will take notes on what's discussed, and do not expect anything to be the final word. If you can't make it, don't hesitate to chime in after the fact!

Wondering about collaborative document editing - I can set up a google doc, but I'm open to other tools. Regardless, I'll write it all up formatted for github and post here.

**Comment by mkborregaard** [comment]
Could be fun to do some other time - closing as stale
